aggregator/handler/common/traffic_controller.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package common
+
+import (
+	"time"
+
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3cluster/kv/util"
+	"github.com/m3db/m3x/instrument"
+	"github.com/m3db/m3x/watch"
+
+	"go.uber.org/atomic"
+)
+
+const (
+	defaultDefaultDisabled = false
+	defaultInitTimeout     = 2 * time.Second
+)
+
+// TrafficControlOptions configurates the traffic control options.
+type TrafficControlOptions interface {
+	// SetStore sets the kv store.
+	SetStore(store kv.Store) TrafficControlOptions
+
+	// Store returns the kv store.
+	Store() kv.Store
+
+	// SetDefaultDisabled sets the default disabled value.
+	SetDefaultDisabled(value bool) TrafficControlOptions
```

Comment:
Would it be more preferable to use `SetDefaultEnabled` instead? I assume you want to rely on the default value of `false` to enable traffic by default, but it's a bit mind-twisting to think about what it means when setting disabled to false.